File: android_webview/test/browser/trafficstats_test_util.cc
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "net/socket/socket_test_util.h"

// Must come after all headers that specialize FromJniType() / ToJniType().
#include "android_webview/test/webview_instrumentation_test_native_jni/TrafficStatsTestUtil_jni.h"

namespace android_webview {

static bool JNI_TrafficStatsTestUtil_CanGetTaggedBytes(JNIEnv* env) {
  return net::CanGetTaggedBytes();
}

static int64_t JNI_TrafficStatsTestUtil_GetTaggedBytes(JNIEnv* env,
                                                       int32_t jexpected_tag) {
  return net::GetTaggedBytes(jexpected_tag);
}

}  // namespace android_webview

DEFINE_JNI(TrafficStatsTestUtil)

File: android_webview/javatests/src/org/chromium/android_webview/test/util/TrafficStatsTestUtil.java
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.android_webview.test.util;

import org.jni_zero.JNINamespace;
import org.jni_zero.NativeMethods;

@JNINamespace("android_webview")
public class TrafficStatsTestUtil {

    public TrafficStatsTestUtil() {}

    /** Returns whether the device supports calling nativeGetTaggedBytes(). */
    public static boolean nativeCanGetTaggedBytes() {
        return TrafficStatsTestUtilJni.get().canGetTaggedBytes();
    }

    /**
     * Query the system to find out how many bytes were received with tag {@code expectedTag} for
     * our UID.
     *
     * @param expectedTag the tag to query for.
     * @return the count of received bytes.
     */
    public static long nativeGetTaggedBytes(int expectedTag) {
        return TrafficStatsTestUtilJni.get().getTaggedBytes(expectedTag);
    }

    @NativeMethods
    interface Natives {
        boolean canGetTaggedBytes();

        long getTaggedBytes(int expectedTag);
    }
}
